### PR TITLE
fix: postgres syntax

### PR DESCRIPTION
**Pull request by YoshkiMatryoshki**
After sleepless nights of research we decided to migrate our database to postgress to provide best posible experience for our customers. This migration leads to some minor errors related to different syntaxex, so for this pull we just fixing post queries without actually touching selection logic. Optimizations and refactorings are planned for next updates.

### CODE REVIEW

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 1]
```diff
@@ -25,13 +25,13 @@ impl Post {
...
             author.last_name AS author_last_name, \
-            GROUP_CONCAT(CONCAT_WS(',', tag.slug, tag.title) SEPARATOR ';') as tags \
+            string_agg(CONCAT_WS(',', tag.slug, tag.title), ';') as tags \
         FROM post \
         LEFT JOIN author ON post.author_id = author.id \
         LEFT JOIN post_tag ON post_tag.post_id  = post.id \
         LEFT JOIN tag ON tag.id = post_tag.tag_id \
         WHERE post.id = #{id} \
-        GROUP BY post.id \
+        GROUP BY post.Id, author.slug, author.first_name, author.last_name \
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:34]: let's use `post.id` to be consistent

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 2]
```diff
@@ -45,13 +45,13 @@ impl Post {
...
             author.last_name AS author_last_name, \
-            GROUP_CONCAT(CONCAT_WS(',', tag.slug, tag.title) SEPARATOR ';') as tags \
+            string_agg(CONCAT_WS(',', tag.slug, tag.title), ';') as tags \
         FROM post \
         LEFT JOIN author ON post.author_id = author.id \
         LEFT JOIN post_tag ON post_tag.post_id  = post.id \
         LEFT JOIN tag ON tag.id = post_tag.tag_id \
         WHERE post.slug = #{slug} \
-        GROUP BY post.id \
+        GROUP BY post.Id, author.slug, author.first_name, author.last_name \
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:54]: let's use `post.id` to be consistent

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 3]
```diff
@@ -65,12 +65,12 @@ impl Post {
...
             author.last_name AS author_last_name, \
-            GROUP_CONCAT(CONCAT_WS(',', tag.slug, tag.title) SEPARATOR ';') as tags \
+            string_agg(concat_ws(',', tag.title, tag.slug), ';') as tags \
         FROM post \
-        LEFT JOIN author ON post.author_id = author.id \
+        JOIN author ON post.author_id = author.id \
         LEFT JOIN post_tag ON post_tag.post_id  = post.id \
         LEFT JOIN tag ON tag.id = post_tag.tag_id \
-        GROUP BY post.id \
+        GROUP BY post.Id, author.slug, author.first_name, author.last_name \
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:73]: let's use `post.id` to be consistent

**Review by tikitko — CHANGES_REQUESTED** [review]

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 1 above)
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:34]:
```suggestion
        GROUP BY post.id, author.slug, author.first_name, author.last_name \
```

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 2 above)
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:54]:
```suggestion
        GROUP BY post.id, author.slug, author.first_name, author.last_name \
```

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 3 above)
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:73]:
```suggestion
        GROUP BY post.id, author.slug, author.first_name, author.last_name \
```

**Review by tikitko — CHANGES_REQUESTED** [review]

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 4]
```diff
@@ -65,12 +65,12 @@ impl Post {
             author.slug AS author_slug, \
             author.first_name AS author_first_name, \
             author.last_name AS author_last_name, \
-            GROUP_CONCAT(CONCAT_WS(',', tag.slug, tag.title) SEPARATOR ';') as tags \
+            string_agg(concat_ws(',', tag.title, tag.slug), ';') as tags \
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:68]: let's use `concat_ws` lowercase for two other cases.

**Review by tikitko — APPROVED** [review]

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 1 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: sure! thank for ur attention! =)

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 4 above)
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:68]: pog

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 2 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: sure! thank for ur attention! =)

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 3 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: sure! thank for ur attention! =)

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 1 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: sure! thank for ur attention! =)

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 4 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: sure! thank for ur attention! =)

Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: champ

**Review by tikitko — APPROVED** [review]
GO POG

**Pull request merged**